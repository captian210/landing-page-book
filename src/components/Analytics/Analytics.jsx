import React, { Component } from 'react';
import PropTypes from 'prop-types';
import ReactGA from 'react-ga';

/**
  ## Overview
  `@front10/landing-page-book/Analytics` is used to send statistics of visits to google analytics
  ## How to use
  1- Add following content to import the component:
  ```js
   import Analytics from "@front10/landing-page-book/dist/components/Analytics";
  ```
  2- Put this code into jsx page:
  ```html
   <Analytics idTracking='UA-0000000-1' />
  ```
 */
class Analytics extends Component {
  constructor(props) {
    super(props);
    ReactGA.initialize(props.idTracking, {
      debug: props.debug,
      testMode: props.testMode
    });
  }

  componentDidMount() {
    const { urlPath } = this.props;
    ReactGA.pageview(urlPath);
  }

  componentDidUpdate(prevProps) {
    ReactGA.pageview(prevProps.urlPath);
  }

  render() {
    return <div />;
  }
}

Analytics.propTypes = {
  /**
   * Tracking ID code of your property in google analytics
   */
  idTracking: PropTypes.string.isRequired,
  /**
   * Path you want to track, for example '/section/page1'. Default route is obtained from the window object
   */
  urlPath: PropTypes.string,
  /**
   * If set to true, will output additional feedback to the console. Default `false`.
   */
  debug: PropTypes.bool,
  /**
   * Enable test mode in the component .Default `false`
   */
  testMode: PropTypes.bool
};
Analytics.defaultProps = {
  urlPath:
    typeof window !== 'undefined' && typeof window.location !== 'undefined'
      ? window.location.pathname + window.location.search
      : '',
  debug: false,
  testMode: false
};

export default Analytics;
