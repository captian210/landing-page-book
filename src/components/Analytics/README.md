<a style="float:right; margin-top: 30px;" target="_blank" href="https://github.com/front10/landing-page-book/edit/master/src/components/Analytics/README.md"> <img width="15px;" src="https://assets-cdn.github.com/images/icons/emoji/unicode/270f.png"/> Edit on Github
</a>

# Analytics

[![standard-readme compliant](https://img.shields.io/badge/standard--readme-OK-green.svg?style=flat-square)](https://github.com/RichardLitt/standard-readme)

## Overview
`@front10/landing-page-book/Analytics` is used to send statistics of visits to google analytics

## How to use
1- Add following content to import the component:

```js
import Analytics from "@front10/landing-page-book/dist/components/Analytics";
```

2- Put this code into jsx page:
```html
<Analytics idTracking='UA-0000000-1' />
```

## Properties:
| </br>Name   | </br>Type | </br>Summary                                                                                 | 
| ------------| - | ------------------------------------------------------------------------------------------------------ |
| idTracking      | `String` | Tracking ID code of your property in google analytics |
| debug      | `Boolean` | If set to true, will output additional feedback to the console. Default `false` |
|  urlPath | `String` | Path you want to track, for example '/section/page1'. Default route is obtained from the window object |
|testMode | `Boolean` | Enable test mode in the component .Default `false` |