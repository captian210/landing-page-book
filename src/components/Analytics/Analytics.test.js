import React from 'react';
import { configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Analytics from './Analytics';
import ReactGA from 'react-ga';

configure({ adapter: new Adapter() });

describe('Analytics', () => {
  let props;
  let mounted;
  const analytics = () => {
    if (!mounted) {
      mounted = mount(<Analytics {...props} />);
    }
    return mounted;
  };
  describe('Analytics', () => {
    beforeEach(() => {
      props = {
        testMode: true,
        idTracking: "U-000000-1"
      };
    });

    it('should render', () => {
      expect(mount(<Analytics {...props} />)).toMatchSnapshot();
    });
  });
});
