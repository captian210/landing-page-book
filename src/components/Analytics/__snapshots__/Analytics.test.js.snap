// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Analytics Analytics should render 1`] = `
ReactWrapper {
  Symbol(enzyme.__unrendered__): <Analytics
    debug={false}
    idTracking="U-000000-1"
    testMode={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__node__): Object {
    "instance": Analytics {
      "_reactInternalFiber": Se {
        "alternate": null,
        "child": Se {
          "alternate": null,
          "child": null,
          "childExpirationTime": 0,
          "effectTag": 0,
          "expirationTime": 0,
          "firstContextDependency": null,
          "firstEffect": null,
          "index": 0,
          "key": null,
          "lastEffect": null,
          "memoizedProps": Object {
            "children": undefined,
            "className": "analytics",
          },
          "memoizedState": null,
          "mode": 0,
          "nextEffect": null,
          "pendingProps": Object {
            "children": undefined,
            "className": "analytics",
          },
          "ref": null,
          "return": [Circular],
          "sibling": null,
          "stateNode": <div
            class="analytics"
          />,
          "tag": 7,
          "type": "div",
          "updateQueue": null,
        },
        "childExpirationTime": 0,
        "effectTag": 1,
        "expirationTime": 0,
        "firstContextDependency": null,
        "firstEffect": null,
        "index": 0,
        "key": null,
        "lastEffect": null,
        "memoizedProps": Object {
          "debug": false,
          "idTracking": "U-000000-1",
          "testMode": true,
        },
        "memoizedState": null,
        "mode": 0,
        "nextEffect": null,
        "pendingProps": Object {
          "debug": false,
          "idTracking": "U-000000-1",
          "testMode": true,
        },
        "ref": null,
        "return": Se {
          "alternate": null,
          "child": [Circular],
          "childExpirationTime": 0,
          "effectTag": 1,
          "expirationTime": 0,
          "firstContextDependency": null,
          "firstEffect": null,
          "index": 0,
          "key": null,
          "lastEffect": null,
          "memoizedProps": Object {
            "Component": [Function],
            "context": null,
            "props": Object {
              "debug": false,
              "idTracking": "U-000000-1",
              "testMode": true,
            },
          },
          "memoizedState": Object {
            "context": null,
            "mount": true,
            "props": Object {
              "debug": false,
              "idTracking": "U-000000-1",
              "testMode": true,
            },
          },
          "mode": 0,
          "nextEffect": null,
          "pendingProps": Object {
            "Component": [Function],
            "context": null,
            "props": Object {
              "debug": false,
              "idTracking": "U-000000-1",
              "testMode": true,
            },
          },
          "ref": null,
          "return": Se {
            "alternate": Se {
              "alternate": [Circular],
              "child": null,
              "childExpirationTime": 0,
              "effectTag": 0,
              "expirationTime": 1,
              "firstContextDependency": null,
              "firstEffect": null,
              "index": 0,
              "key": null,
              "lastEffect": null,
              "memoizedProps": null,
              "memoizedState": null,
              "mode": 0,
              "nextEffect": null,
              "pendingProps": null,
              "ref": null,
              "return": null,
              "sibling": null,
              "stateNode": Object {
                "containerInfo": <div>
                  <div
                    class="analytics"
                  />
                </div>,
                "context": Object {},
                "current": [Circular],
                "didError": false,
                "earliestPendingTime": 0,
                "earliestSuspendedTime": 0,
                "expirationTime": 0,
                "finishedWork": null,
                "firstBatch": null,
                "hydrate": false,
                "latestPendingTime": 0,
                "latestPingedTime": 0,
                "latestSuspendedTime": 0,
                "nextExpirationTimeToWorkOn": 0,
                "nextScheduledRoot": null,
                "pendingChildren": null,
                "pendingCommitExpirationTime": 0,
                "pendingContext": null,
                "timeoutHandle": -1,
              },
              "tag": 5,
              "type": null,
              "updateQueue": Object {
                "baseState": null,
                "firstCapturedEffect": null,
                "firstCapturedUpdate": null,
                "firstEffect": null,
                "firstUpdate": Object {
                  "callback": null,
                  "expirationTime": 1,
                  "next": null,
                  "nextEffect": null,
                  "payload": Object {
                    "element": <WrapperComponent
                      Component={[Function]}
                      context={null}
                      props={
                        Object {
                          "debug": false,
                          "idTracking": "U-000000-1",
                          "testMode": true,
                        }
                      }
                    />,
                  },
                  "tag": 0,
                },
                "lastCapturedEffect": null,
                "lastCapturedUpdate": null,
                "lastEffect": null,
                "lastUpdate": Object {
                  "callback": null,
                  "expirationTime": 1,
                  "next": null,
                  "nextEffect": null,
                  "payload": Object {
                    "element": <WrapperComponent
                      Component={[Function]}
                      context={null}
                      props={
                        Object {
                          "debug": false,
                          "idTracking": "U-000000-1",
                          "testMode": true,
                        }
                      }
                    />,
                  },
                  "tag": 0,
                },
              },
            },
            "child": [Circular],
            "childExpirationTime": 0,
            "effectTag": 32,
            "expirationTime": 0,
            "firstContextDependency": null,
            "firstEffect": [Circular],
            "index": 0,
            "key": null,
            "lastEffect": [Circular],
            "memoizedProps": null,
            "memoizedState": Object {
              "element": <WrapperComponent
                Component={[Function]}
                context={null}
                props={
                  Object {
                    "debug": false,
                    "idTracking": "U-000000-1",
                    "testMode": true,
                  }
                }
              />,
            },
            "mode": 0,
            "nextEffect": null,
            "pendingProps": null,
            "ref": null,
            "return": null,
            "sibling": null,
            "stateNode": Object {
              "containerInfo": <div>
                <div
                  class="analytics"
                />
              </div>,
              "context": Object {},
              "current": [Circular],
              "didError": false,
              "earliestPendingTime": 0,
              "earliestSuspendedTime": 0,
              "expirationTime": 0,
              "finishedWork": null,
              "firstBatch": null,
              "hydrate": false,
              "latestPendingTime": 0,
              "latestPingedTime": 0,
              "latestSuspendedTime": 0,
              "nextExpirationTimeToWorkOn": 0,
              "nextScheduledRoot": null,
              "pendingChildren": null,
              "pendingCommitExpirationTime": 0,
              "pendingContext": null,
              "timeoutHandle": -1,
            },
            "tag": 5,
            "type": null,
            "updateQueue": Object {
              "baseState": Object {
                "element": <WrapperComponent
                  Component={[Function]}
                  context={null}
                  props={
                    Object {
                      "debug": false,
                      "idTracking": "U-000000-1",
                      "testMode": true,
                    }
                  }
                />,
              },
              "firstCapturedEffect": null,
              "firstCapturedUpdate": null,
              "firstEffect": null,
              "firstUpdate": null,
              "lastCapturedEffect": null,
              "lastCapturedUpdate": null,
              "lastEffect": null,
              "lastUpdate": null,
            },
          },
          "sibling": null,
          "stateNode": WrapperComponent {
            "_reactInternalFiber": [Circular],
            "context": Object {},
            "props": Object {
              "Component": [Function],
              "context": null,
              "props": Object {
                "debug": false,
                "idTracking": "U-000000-1",
                "testMode": true,
              },
            },
            "refs": Object {},
            "state": Object {
              "context": null,
              "mount": true,
              "props": Object {
                "debug": false,
                "idTracking": "U-000000-1",
                "testMode": true,
              },
            },
            "updater": Object {
              "enqueueForceUpdate": [Function],
              "enqueueReplaceState": [Function],
              "enqueueSetState": [Function],
              "isMounted": [Function],
            },
          },
          "tag": 2,
          "type": [Function],
          "updateQueue": null,
        },
        "sibling": null,
        "stateNode": [Circular],
        "tag": 2,
        "type": [Function],
        "updateQueue": null,
      },
      "context": Object {},
      "props": Object {
        "debug": false,
        "idTracking": "U-000000-1",
        "testMode": true,
      },
      "refs": Object {},
      "state": null,
      "updater": Object {
        "enqueueForceUpdate": [Function],
        "enqueueReplaceState": [Function],
        "enqueueSetState": [Function],
        "isMounted": [Function],
      },
    },
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "debug": false,
      "idTracking": "U-000000-1",
      "testMode": true,
    },
    "ref": null,
    "rendered": Object {
      "instance": <div
        class="analytics"
      />,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": undefined,
        "className": "analytics",
      },
      "ref": null,
      "rendered": Array [
        undefined,
      ],
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": Analytics {
        "_reactInternalFiber": Se {
          "alternate": null,
          "child": Se {
            "alternate": null,
            "child": null,
            "childExpirationTime": 0,
            "effectTag": 0,
            "expirationTime": 0,
            "firstContextDependency": null,
            "firstEffect": null,
            "index": 0,
            "key": null,
            "lastEffect": null,
            "memoizedProps": Object {
              "children": undefined,
              "className": "analytics",
            },
            "memoizedState": null,
            "mode": 0,
            "nextEffect": null,
            "pendingProps": Object {
              "children": undefined,
              "className": "analytics",
            },
            "ref": null,
            "return": [Circular],
            "sibling": null,
            "stateNode": <div
              class="analytics"
            />,
            "tag": 7,
            "type": "div",
            "updateQueue": null,
          },
          "childExpirationTime": 0,
          "effectTag": 1,
          "expirationTime": 0,
          "firstContextDependency": null,
          "firstEffect": null,
          "index": 0,
          "key": null,
          "lastEffect": null,
          "memoizedProps": Object {
            "debug": false,
            "idTracking": "U-000000-1",
            "testMode": true,
          },
          "memoizedState": null,
          "mode": 0,
          "nextEffect": null,
          "pendingProps": Object {
            "debug": false,
            "idTracking": "U-000000-1",
            "testMode": true,
          },
          "ref": null,
          "return": Se {
            "alternate": null,
            "child": [Circular],
            "childExpirationTime": 0,
            "effectTag": 1,
            "expirationTime": 0,
            "firstContextDependency": null,
            "firstEffect": null,
            "index": 0,
            "key": null,
            "lastEffect": null,
            "memoizedProps": Object {
              "Component": [Function],
              "context": null,
              "props": Object {
                "debug": false,
                "idTracking": "U-000000-1",
                "testMode": true,
              },
            },
            "memoizedState": Object {
              "context": null,
              "mount": true,
              "props": Object {
                "debug": false,
                "idTracking": "U-000000-1",
                "testMode": true,
              },
            },
            "mode": 0,
            "nextEffect": null,
            "pendingProps": Object {
              "Component": [Function],
              "context": null,
              "props": Object {
                "debug": false,
                "idTracking": "U-000000-1",
                "testMode": true,
              },
            },
            "ref": null,
            "return": Se {
              "alternate": Se {
                "alternate": [Circular],
                "child": null,
                "childExpirationTime": 0,
                "effectTag": 0,
                "expirationTime": 1,
                "firstContextDependency": null,
                "firstEffect": null,
                "index": 0,
                "key": null,
                "lastEffect": null,
                "memoizedProps": null,
                "memoizedState": null,
                "mode": 0,
                "nextEffect": null,
                "pendingProps": null,
                "ref": null,
                "return": null,
                "sibling": null,
                "stateNode": Object {
                  "containerInfo": <div>
                    <div
                      class="analytics"
                    />
                  </div>,
                  "context": Object {},
                  "current": [Circular],
                  "didError": false,
                  "earliestPendingTime": 0,
                  "earliestSuspendedTime": 0,
                  "expirationTime": 0,
                  "finishedWork": null,
                  "firstBatch": null,
                  "hydrate": false,
                  "latestPendingTime": 0,
                  "latestPingedTime": 0,
                  "latestSuspendedTime": 0,
                  "nextExpirationTimeToWorkOn": 0,
                  "nextScheduledRoot": null,
                  "pendingChildren": null,
                  "pendingCommitExpirationTime": 0,
                  "pendingContext": null,
                  "timeoutHandle": -1,
                },
                "tag": 5,
                "type": null,
                "updateQueue": Object {
                  "baseState": null,
                  "firstCapturedEffect": null,
                  "firstCapturedUpdate": null,
                  "firstEffect": null,
                  "firstUpdate": Object {
                    "callback": null,
                    "expirationTime": 1,
                    "next": null,
                    "nextEffect": null,
                    "payload": Object {
                      "element": <WrapperComponent
                        Component={[Function]}
                        context={null}
                        props={
                          Object {
                            "debug": false,
                            "idTracking": "U-000000-1",
                            "testMode": true,
                          }
                        }
                      />,
                    },
                    "tag": 0,
                  },
                  "lastCapturedEffect": null,
                  "lastCapturedUpdate": null,
                  "lastEffect": null,
                  "lastUpdate": Object {
                    "callback": null,
                    "expirationTime": 1,
                    "next": null,
                    "nextEffect": null,
                    "payload": Object {
                      "element": <WrapperComponent
                        Component={[Function]}
                        context={null}
                        props={
                          Object {
                            "debug": false,
                            "idTracking": "U-000000-1",
                            "testMode": true,
                          }
                        }
                      />,
                    },
                    "tag": 0,
                  },
                },
              },
              "child": [Circular],
              "childExpirationTime": 0,
              "effectTag": 32,
              "expirationTime": 0,
              "firstContextDependency": null,
              "firstEffect": [Circular],
              "index": 0,
              "key": null,
              "lastEffect": [Circular],
              "memoizedProps": null,
              "memoizedState": Object {
                "element": <WrapperComponent
                  Component={[Function]}
                  context={null}
                  props={
                    Object {
                      "debug": false,
                      "idTracking": "U-000000-1",
                      "testMode": true,
                    }
                  }
                />,
              },
              "mode": 0,
              "nextEffect": null,
              "pendingProps": null,
              "ref": null,
              "return": null,
              "sibling": null,
              "stateNode": Object {
                "containerInfo": <div>
                  <div
                    class="analytics"
                  />
                </div>,
                "context": Object {},
                "current": [Circular],
                "didError": false,
                "earliestPendingTime": 0,
                "earliestSuspendedTime": 0,
                "expirationTime": 0,
                "finishedWork": null,
                "firstBatch": null,
                "hydrate": false,
                "latestPendingTime": 0,
                "latestPingedTime": 0,
                "latestSuspendedTime": 0,
                "nextExpirationTimeToWorkOn": 0,
                "nextScheduledRoot": null,
                "pendingChildren": null,
                "pendingCommitExpirationTime": 0,
                "pendingContext": null,
                "timeoutHandle": -1,
              },
              "tag": 5,
              "type": null,
              "updateQueue": Object {
                "baseState": Object {
                  "element": <WrapperComponent
                    Component={[Function]}
                    context={null}
                    props={
                      Object {
                        "debug": false,
                        "idTracking": "U-000000-1",
                        "testMode": true,
                      }
                    }
                  />,
                },
                "firstCapturedEffect": null,
                "firstCapturedUpdate": null,
                "firstEffect": null,
                "firstUpdate": null,
                "lastCapturedEffect": null,
                "lastCapturedUpdate": null,
                "lastEffect": null,
                "lastUpdate": null,
              },
            },
            "sibling": null,
            "stateNode": WrapperComponent {
              "_reactInternalFiber": [Circular],
              "context": Object {},
              "props": Object {
                "Component": [Function],
                "context": null,
                "props": Object {
                  "debug": false,
                  "idTracking": "U-000000-1",
                  "testMode": true,
                },
              },
              "refs": Object {},
              "state": Object {
                "context": null,
                "mount": true,
                "props": Object {
                  "debug": false,
                  "idTracking": "U-000000-1",
                  "testMode": true,
                },
              },
              "updater": Object {
                "enqueueForceUpdate": [Function],
                "enqueueReplaceState": [Function],
                "enqueueSetState": [Function],
                "isMounted": [Function],
              },
            },
            "tag": 2,
            "type": [Function],
            "updateQueue": null,
          },
          "sibling": null,
          "stateNode": [Circular],
          "tag": 2,
          "type": [Function],
          "updateQueue": null,
        },
        "context": Object {},
        "props": Object {
          "debug": false,
          "idTracking": "U-000000-1",
          "testMode": true,
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
        },
      },
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "debug": false,
        "idTracking": "U-000000-1",
        "testMode": true,
      },
      "ref": null,
      "rendered": Object {
        "instance": <div
          class="analytics"
        />,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": "analytics",
        },
        "ref": null,
        "rendered": Array [
          undefined,
        ],
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
